plugins/serializers/nowmetric/nowmetric_test.go

```diff
@@ -0,0 +1,184 @@
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle": int64(90),
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s, _ := NewSerializer(0)
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	expS := []byte(fmt.Sprintf(`[{"metric_type":"usage_idle","resource":"","node":"","value":90,"timestamp":%d,"ci2metric_id":null,"source":"Telegraf"}]`, (now.UnixNano() / int64(time.Millisecond))))
+	assert.Equal(t, string(expS), string(buf))
+}
+
+func TestSerializeMetricString(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle": "foobar",
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s, _ := NewSerializer(0)
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "null", string(buf))
+}
+
+func TestSerializeMultiFields(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle":  int64(90),
+		"usage_total": 8559615,
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s, _ := NewSerializer(0)
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	expS := []byte(fmt.Sprintf(`[{"metric_type":"usage_idle","resource":"","node":"","value":90,"timestamp":%d,"ci2metric_id":null,"source":"Telegraf"},{"metric_type":"usage_total","resource":"","node":"","value":8559615,"timestamp":%d,"ci2metric_id":null,"source":"Telegraf"}]`, (now.UnixNano() / int64(time.Millisecond)), (now.UnixNano() / int64(time.Millisecond))))
+	assert.Equal(t, string(expS), string(buf))
+}
+
+func TestSerializeMetricWithEscapes(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu tag": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"U,age=Idle": int64(90),
+	}
+	m, err := metric.New("My CPU", tags, fields, now)
+	assert.NoError(t, err)
+
+	s, _ := NewSerializer(0)
+	buf, err := s.Serialize(m)
+	assert.NoError(t, err)
+
+	expS := []byte(fmt.Sprintf(`[{"metric_type":"U,age=Idle","resource":"","node":"","value":90,"timestamp":%d,"ci2metric_id":null,"source":"Telegraf"}]`, (now.UnixNano() / int64(time.Millisecond))))
+	assert.Equal(t, string(expS), string(buf))
+}
+
+func TestSerializeBatch(t *testing.T) {
+	m := MustMetric(
+		metric.New(
+			"cpu",
+			map[string]string{},
+			map[string]interface{}{
+				"value": 42.0,
+			},
+			time.Unix(0, 0),
+		),
+	)
+
+	metrics := []telegraf.Metric{m, m}
+	s, _ := NewSerializer(0)
+	buf, err := s.SerializeBatch(metrics)
+	require.NoError(t, err)
+	require.Equal(t, []byte(`[{"metric_type":"value","resource":"","node":"","value":42,"timestamp":0,"ci2metric_id":null,"source":"Telegraf"}][{"metric_type":"value","resource":"","node":"","value":42,"timestamp":0,"ci2metric_id":null,"source":"Telegraf"}]`), buf)
```

Comment:
This does not seem correct to me, shouldn't the metrics all be in one array: `[{}, {}]` instead of `[{}][{}]`?

Reply:
I think the SerializeBatch function is not producing the output the servicenow API requires.  The current output with multiple arrays is not even valid JSON: ` [{}][{}]`.  I would expect to output multiple objects within a single array like: `[{}, {}]`.

Reply:
Ok i get your point & I need your advice for this one..
I reproduced what i have seen for other serializer, but this test is not correct.  

SerializeBatch is expecting a metric array as parameter...
As this function just loop over the array and call the marshal function for each metric, it results in this string.

What would you recommend?

a. Should i merge the metric array in one metric object before marshalling?
b. Or Modify the SerializeBatch test, to ingest only one metric object but with multiples metric.FieldList() ?

I have tested the actual code on both Windows and Linux platform and I'm able to ingest all metrics on ServiceNow side successfully. So i assume the issue is "limited" to the test function.

Reply:
For which use case or situation is the SerializeBatch function called?

Reply:
I mean I'm testing my agent with Windows Performance Counters plugin setup and against a standard AWS Linux box then i compare the metric stream in ServiceNow and the file output. 
Is SerializeBatch invoked in these test case? I have the feeling that for both case, it's the Serialize function which is invoked. How can SerializeBatch be invoked?

Reply:
It is up to the output plugin to choose between Serialize and SerializeBatch, of interest though is that the HTTP output uses SerializeBatch while the file output uses Serialize.

Reply:
Ok. This makes it much more clear…. Thanks.
The MID listener didn't returned any wrong code with this payload, but some metrics got probably ignored.
I have modified the SerializeBatch function and I'm currently running a bunch of tests. They look positive so far.

Continuous-integration tests are failing outside of my code. Is this an issue on my side?

Reply:
Except continuous integration tests, the functional one are ok now.
Let me know how to fix others.
Thanks for your help